File: scala/src/main/scala/com/spoonacular/client/model/GetRandomRecipes200ResponseRecipesInner.scala
package com.spoonacular.client.model

import io.circe._
import io.finch.circe._
import io.circe.generic.semiauto._
import io.circe.java8.time._
import org.openapitools._
import com.spoonacular.client.model.BigDecimal
import com.spoonacular.client.model.GetRecipeInformation200ResponseExtendedIngredientsInner
import com.spoonacular.client.model.GetRecipeInformation200ResponseWinePairing
import scala.collection.immutable.Seq

/**
 * 
 * @param id 
 * @param title 
 * @param image 
 * @param imageType 
 * @param servings 
 * @param readyInMinutes 
 * @param license 
 * @param sourceName 
 * @param sourceUrl 
 * @param spoonacularSourceUrl 
 * @param aggregateLikes 
 * @param healthScore 
 * @param spoonacularScore 
 * @param pricePerServing 
 * @param analyzedInstructions 
 * @param cheap 
 * @param creditsText 
 * @param cuisines 
 * @param dairyFree 
 * @param diets 
 * @param gaps 
 * @param glutenFree 
 * @param instructions 
 * @param ketogenic 
 * @param lowFodmap 
 * @param occasions 
 * @param sustainable 
 * @param vegan 
 * @param vegetarian 
 * @param veryHealthy 
 * @param veryPopular 
 * @param whole30 
 * @param weightWatcherSmartPoints 
 * @param dishTypes 
 * @param extendedIngredients 
 * @param summary 
 * @param winePairing 
 */
case class GetRandomRecipes200ResponseRecipesInner(id: Int,
                title: String,
                image: String,
                imageType: String,
                servings: BigDecimal,
                readyInMinutes: Int,
                license: String,
                sourceName: String,
                sourceUrl: String,
                spoonacularSourceUrl: String,
                aggregateLikes: BigDecimal,
                healthScore: BigDecimal,
                spoonacularScore: BigDecimal,
                pricePerServing: BigDecimal,
                analyzedInstructions: Option[Seq[Object]],
                cheap: Boolean,
                creditsText: String,
                cuisines: Option[Seq[String]],
                dairyFree: Boolean,
                diets: Option[Seq[String]],
                gaps: String,
                glutenFree: Boolean,
                instructions: String,
                ketogenic: Boolean,
                lowFodmap: Boolean,
                occasions: Option[Seq[String]],
                sustainable: Boolean,
                vegan: Boolean,
                vegetarian: Boolean,
                veryHealthy: Boolean,
                veryPopular: Boolean,
                whole30: Boolean,
                weightWatcherSmartPoints: BigDecimal,
                dishTypes: Option[Seq[String]],
                extendedIngredients: Option[Set[GetRecipeInformation200ResponseExtendedIngredientsInner]],
                summary: String,
                winePairing: Option[GetRecipeInformation200ResponseWinePairing]
                )

object GetRandomRecipes200ResponseRecipesInner {
    /**
     * Creates the codec for converting GetRandomRecipes200ResponseRecipesInner from and to JSON.
     */
    implicit val decoder: Decoder[GetRandomRecipes200ResponseRecipesInner] = deriveDecoder
    implicit val encoder: ObjectEncoder[GetRandomRecipes200ResponseRecipesInner] = deriveEncoder
}

File: clojure/src/spoonacular_api/specs/get_random_recipes_200_response_recipes_inner.clj
(ns spoonacular-api.specs.get-random-recipes-200-response-recipes-inner
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [spoonacular-api.specs.get-recipe-information-200-response-extended-ingredients-inner :refer :all]
            [spoonacular-api.specs.get-recipe-information-200-response-wine-pairing :refer :all]
            )
  (:import (java.io File)))


(def get-random-recipes-200-response-recipes-inner-data
  {
   (ds/req :id) int?
   (ds/req :title) string?
   (ds/req :image) string?
   (ds/req :imageType) string?
   (ds/req :servings) float?
   (ds/req :readyInMinutes) int?
   (ds/req :license) string?
   (ds/req :sourceName) string?
   (ds/req :sourceUrl) string?
   (ds/req :spoonacularSourceUrl) string?
   (ds/req :aggregateLikes) float?
   (ds/req :healthScore) float?
   (ds/req :spoonacularScore) float?
   (ds/req :pricePerServing) float?
   (ds/opt :analyzedInstructions) (s/coll-of any?)
   (ds/req :cheap) boolean?
   (ds/req :creditsText) string?
   (ds/opt :cuisines) (s/coll-of string?)
   (ds/req :dairyFree) boolean?
   (ds/opt :diets) (s/coll-of string?)
   (ds/req :gaps) string?
   (ds/req :glutenFree) boolean?
   (ds/req :instructions) string?
   (ds/req :ketogenic) boolean?
   (ds/req :lowFodmap) boolean?
   (ds/opt :occasions) (s/coll-of string?)
   (ds/req :sustainable) boolean?
   (ds/req :vegan) boolean?
   (ds/req :vegetarian) boolean?
   (ds/req :veryHealthy) boolean?
   (ds/req :veryPopular) boolean?
   (ds/req :whole30) boolean?
   (ds/req :weightWatcherSmartPoints) float?
   (ds/opt :dishTypes) (s/coll-of string?)
   (ds/opt :extendedIngredients) (s/coll-of get-recipe-information-200-response-extended-ingredients-inner-spec)
   (ds/req :summary) string?
   (ds/opt :winePairing) get-recipe-information-200-response-wine-pairing-spec
   })

(def get-random-recipes-200-response-recipes-inner-spec
  (ds/spec
    {:name ::get-random-recipes-200-response-recipes-inner
     :spec get-random-recipes-200-response-recipes-inner-data}))
